webots_ros2_driver/webots_ros2_driver/urdf_spawner.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+
+# Copyright 1996-2022 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This process simply sends urdf information to the Spawner through a service."""
+
+from launch.actions import ExecuteProcess
+
+
+def get_webots_driver_node(event, driver_node):
+    """Return the driver node in case the service response is successful."""
+    if "success=True" in event.text.decode().strip():
+        return driver_node
+    print("WARNING: the Ros2Supervisor was not able to spawn this URDF robot.")
+    return
+
+class URDFSpawner(ExecuteProcess):
+    def __init__(self, output='log', name=None, robot_description=None, translation='0 0 0', rotation='0 0 1 0', normal=False, box_collision=False, init_pos=None, **kwargs):
+        message = '{robot: {'
+
+        if name:
+            message += 'name: "' + name + '",'
+        if robot_description:
+            # Prepare the robot_description to be send via command.
+            robot_description = robot_description.replace("'", "\'")
+            robot_description = robot_description.replace('"', '\\"')
```

Comment:
I will double check but as my string is using `"` quotes as delimiters I think I have to do it like this, otherwise the `"` contained in the robot description won't be different than the `"` at the end of the string.

Reply:
OK, thank you for checking. My proposed change still remains a valid simplification.

Reply:
Thanks for the reminder, I was focus on the question of the `\\"` !